lib/utils.pm

```diff
@@ -891,6 +892,29 @@ sub service_action {
     }
 }
 
+=head2 install_all_from_repo
+   will install all packages in repo defined by INSTALL_ALL_REPO variable
+   with ability to exclude some of them by using INSTALL_ALL_EXCEPT which suppose to contain
+   space separated list of packages
+=cut
+sub install_all_from_repo {
+    my $repo     = get_required_var('INSTALL_ALL_REPO');
+    my $grep_str = "";
+    if (get_var('INSTALL_ALL_EXCEPT')) {
+        #spliting space separated list of packages into array to iterate over it
+        my @packages_array = split(/ /, get_var('INSTALL_ALL_EXCEPT'));
+        my $grep_regex = "";
+        #loop which will build regex from package names array in form "package1$|package2$|package3$" expected by grep -vE
+        foreach my $package (@packages_array) {
+            if (length $grep_regex) { $grep_regex .= "|" };
+            $grep_regex .= sprintf("$package\$");
+        }
+        $grep_str = sprintf("|grep -vE \"(%s)\"", $grep_regex);
```

Comment:
actually I need '$' after every word , because I don't want to filter out "package1-devel" when I specified "package1" in a list